### DISCUSSION TITLE
YOLOV5

### DISCUSSION THREAD

**Discussion by Maryasegere** [community/community#19358] [Other Feature Feedback, Questions, & Ideas]
How can i add social distancing on the original yolov5?

**Comment by bossoff** [comment]
@Maryasegere I think this can only be decided by GitHub only. Keep on merging based on needed dependency and all.

**Reply by ertugrul37** [reply]
kesinlikle

**Comment by thuongtruong1009** [comment]
Hi @Maryasegere 
You can try with https://towardsai.net/p/computer-vision/yolo-v5-object-detection-on-a-custom-dataset

**Comment by ertugrul37** [comment]
> Orijinal yolov5'e sosyal mesafeyi nasıl ekleyebilirim?
@Maryasegere 

-Open your repository;
-Create new branch;
-Do some updates;
-Then, create pull request to main branch;
-IMPORTANT! Add reviewer. To do this, you need to add access to repository someone, then add him as reviewer;
-After this, just click merge as you do it always
-Check your profile and see YOLO achievement.

**Comment by rilysh** [comment]
> How can i add social distancing on the original yolov5?

https://github.com/ultralytics/yolov5

**Comment by mpouziotasd** [comment]
Instead of providing you with a link, I'll gladly provide you with a simple answer. Your best method to add "social distancing", is with simple tracking techniques using two or more objects to calculate the distance from the center of your bounding boxes. Of course this might not always be the most efficient method, as this method does not take into consideration the depth of the video/image.

If you're interested in doing some math yourself you could try and calculate the Pythagorean distance using the centers of the bounding boxes, and a threshold distance value determined by the user. If the distance between the two bounding boxes is smaller than the threshold value, then you could do the following:
You could change the bounding box color to say... red from green or the name of the class itself Person - Safe, to Person - Not safe

This solution is a simple implementation not necessarily made for realistic problems, realistic problems tend to use trackers or more complex mathematics.
It does sound like a fun project! Several implementations like this have also been introduced online, including on YouTube.
